### DISCUSSION TITLE
Execution of workbench closing / session tests

### DISCUSSION THREAD

**Discussion by HeikoKlare** [eclipse-platform/eclipse.platform.ui#1143] [Q&A]
When adding a test for #1086 that needs to close the workbench, I found two kinds of tests that close/restart a new workbench, which are those executed as `SessionTest` in the `org.eclipse.ui.tests` bundle or those executed as ordinary tests with each starting its own workbench in the `org.eclipse.ui.tests.rcp` bundle. Now all these tests do not seem to be executed in any CI environment (neither GitHub actions nor in Jenkins builds). 

The session tests are disabled in Maven build used for GH actions due to only executing one test suite that does not include them https://github.com/eclipse-platform/eclipse.platform.ui/blob/9739280731cd106f794a857d463d422a015b39d1/tests/org.eclipse.ui.tests/build.properties#L32 and in Jenkins integration build due to https://github.com/eclipse-platform/eclipse.platform.ui/blob/9739280731cd106f794a857d463d422a015b39d1/tests/org.eclipse.ui.tests/test.xml#L69-L74

The RCP tests are disabled in Maven build used for GH actions due to tests being skipped completely https://github.com/eclipse-platform/eclipse.platform.ui/blob/9739280731cd106f794a857d463d422a015b39d1/tests/org.eclipse.ui.tests.rcp/build.properties#L30 and the bundle not being considered in Jenkins integration builds (at least I cannot find any logs for it, e.g., [here](https://download.eclipse.org/eclipse/downloads/drops4/I20230919-1800/logs.php#console)).

Does anyone know whether...
1. these tests are only not executed in automated builds for technical reasons and would be worth some effort to be enabled there?
2. there are other kinds of tests that allow to test the closing behavior of a workbench with which a new test could/should be aligned?

Thanks in advance!

**Comment by iloveeclipse** [comment]
See https://github.com/eclipse-platform/eclipse.platform.ui/commit/a5dc9ce8304b69a7fe2774d6d3ac1d307fb5d164 / https://bugs.eclipse.org/bugs/show_bug.cgi?id=398869. The tests were disabled during transition to e4 (because lot of things were broken I guess), and like with many other bad things happened during this transition, they were not enabled anymore.

**Reply by HeikoKlare** [reply]
Thanks for that information!
At least, it seems to be possible to easily fix and reenable the RCP test project, see #1145. I will provide the reactivation of those tests as proper PRs to ensure that we have some testing for behavior of workbench sessions (including startup/shutdown) again.
